backend/src/question-service/controllers/questions.ts

```diff
@@ -99,9 +99,65 @@ export const deleteQuestion: RequestHandler[] = [
     res.sendStatus(200);
   },
 ];
-// TODO: updateQuestion
+
 export const getQuestionCategories: RequestHandler = (req, res) => {
   res.json({ data: categoryEnum });
 };
 
-// TODO: updateQuestion, deleteQuestion
+// Submitting edit form 
+export const updateQuestion: RequestHandler[] = [
+  
+  param("questionId").notEmpty().withMessage("questionId field cannot be empty."),
+  param("questionId").isNumeric().withMessage("questionId should be a number."),
+  body("title").notEmpty().trim().withMessage("title cannot be empty."),
+  body("categories").isArray().withMessage("categories should be an array."),
+  body("complexity")
+    .isIn(complexityEnum)
+    .withMessage(`complexity should be one of ${complexityEnum.join(", ")}.`),
+  body("linkToQuestion")
+    .isURL()
+    .withMessage("linkToQuestion should be a valid URL."),
+  body("questionDescription")
+    .notEmpty()
+    .withMessage("questionDescription should not be empty."),
+  
+  async (req, res) => {  
+    // validation errors for both above
+    if (!validationResult(req).isEmpty()) {
+      res.status(400).json({ errors: validationResult(req).array() });
+      return;
+    }
+
+    // Find question
+    const questionId = req.params.questionId;
+    const existingQuestion = await QuestionModel.findOne({
+      questionID: questionId,
+    });
+    // catch error if no existing question
+    if (!existingQuestion) {
+      res.status(400).json({ errors: [{ msg: "question does not exist." }] });
+      return;
+    }
+
+    const formData = matchedData(req);
+  
+    // Check if a question with similar title (case insensitive) is already added.
+    const sameQuestionExists = await QuestionModel.exists({
+      title: { $regex: new RegExp("^" + formData.title + "$"), $options: "i" },
+    });
+    if (sameQuestionExists) {
+      res.status(400).json({
+        errors: [{ msg: "Question already exists in the database." }],
+      });
+      return;
+    }
```

Comment:
for update, users should be able to update the complexity/ category/ link to question, while keeping the title the same?

currently, if the title remains the same and other fields change, it's considered invalid edit

Reply:
Addressed in next commit. Checks for duplicate title, while still allowing for same title with other changes in other fields.